File: resources/scop_1.5.1/examples/sos.cpp
// sos.cpp - DMI - 3-1-2002

#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>
#include <syslog.h>

#include <sys/types.h>
#include <sys/socket.h>

#include <scop.h>

int main(int argc, char **argv)
{
	int sock;
	char *buf;
	int rpc_flag;

	sock = scop_open("localhost", "sos", 1);
	if(sock == -1)
	{
		printf("Can't connect to scopserver.\n");
		exit(0);
	}
	
	if(fork() > 0) exit(0); // Detach
	
	while(1)
	{
		buf = scop_get_message(sock);
		if(buf == NULL)
		{
			syslog(LOG_INFO, "Lost connection to scopserver.\n");
			exit(0);
		}
		syslog(LOG_INFO, "%s\n", buf);
		delete[] buf;
	}
	
	close(sock);
	return 0;
}

File: resources/scop_1.5.1/java/examples/SOS.java
// SOS.java - DMI - 27-25-02

import java.io.*;

/* Unlike the C++ version, we don't log messages to syslog or detach
	to run in the background... so this is a fairly boring example! */

public class SOS
{
	public static void main(String[] args)
	{
		SCOP scop;
		String buf;
		int rpc_flag;

		scop = new SCOP("localhost", "SOS", true);
		if(scop.connection_ok() == false)
		{
			System.out.println("Can't connect to scopserver.");
			System.exit(0);
		}

		while(true)
		{
			buf = scop.get_message();
			if(buf == null)
			{
				System.out.println("Lost connection to scopserver.");
				System.exit(0);
			}
			System.out.println(buf);
		}
		// scop.close();
	}
}
